### PR TITLE
updated protocol, add some new pets, fix level points bug, add custom pet

### PR DESCRIPTION
**Pull request by edwinyoo44**
updated protocol to 440 and add some new pets

### CODE REVIEW

File: src/BlockHorizons/BlockPets/listeners/EventListener.php [hunk 1]
```diff
@@ -97,10 +97,12 @@ function(array $petData) use($player, $loader): void {
 							(bool) $data["Visible"],
 							$data["Inventory"]
 						);
-						$pet->spawnToAll();
-						$pet->setDormant(false);
-						if($hard_reset) {
-							$pet->close();
+						if(!is_null($pet)){
```
Comment (lukeeey) [inline src/BlockHorizons/BlockPets/listeners/EventListener.php:100]: if($pet !== null)

Comment (AIPTU) [reply src/BlockHorizons/BlockPets/listeners/EventListener.php:100]: How about issue when pet dies? a lot of people have a problem with that

Comment (lukeeey) [reply src/BlockHorizons/BlockPets/listeners/EventListener.php:100]: I have no idea i haven't had the time to look at it. I fixed the majority of bugs so the plugin was usable

**Comment by edwinyoo44** [comment]
Fix when feeding or getting level points, the level will not increase
#173 #158 #205 #151 #222

File: src/BlockHorizons/BlockPets/listeners/EventListener.php (hunk 1 above)
Comment (edwinyoo44) [reply src/BlockHorizons/BlockPets/listeners/EventListener.php]: https://github.com/BlockHorizons/BlockPets/pull/231/commits/f9672a12db7d4e1f7e06ec9a8d1c622f1c7a8b71

**Comment by edwinyoo44** [comment]
Add #182 Custom pets without changing existing data

Usage: /spawnpet CustomWalking.<type>

Example: /spawnpet CustomWalking.campfire

Reminder, please make sure that the SKIN file with the corresponding name is in resources\CustomPet

**Comment by lukeeey** [comment]
I’ll review this soon

**Comment by Ifera** [comment]
Thanks for the PR but closing this as its very much outdated now.

**Comment by edwinyoo44** [comment]
> Thanks for the PR but closing this as its very much outdated now.

Thanks, I will try to remake the CustomPet

**Pull request closed without merging**